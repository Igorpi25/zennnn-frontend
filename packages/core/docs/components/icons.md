# Icons
The `Icon` component.

## Usage
<Icon>M13 11V6h-2v5H6v2h5v5h2v-5h5v-2h-5z</Icon>

---

```vue
<Icon>
  M13 11V6h-2v5H6v2h5v5h2v-5h5v-2h-5z
</Icon>
```

## Api
- [Icon](../api/icon)

## Sizes
<Example file="Icon/Sizes" />

## Clickable
<Example file="Icon/Clickable" />

## List
<Example file="Icon/List" />
