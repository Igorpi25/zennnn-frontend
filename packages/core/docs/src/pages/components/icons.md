# Icons
The `Icon` component.

## Usage
<Example file="Icon/Usage" />

## Api
- <router-link to="/api/icon">Icon</router-link>

## Sizes
<Example file="Icon/Sizes" />

## Clickable
<Example file="Icon/Clickable" />

## List
<Example file="Icon/List" />
