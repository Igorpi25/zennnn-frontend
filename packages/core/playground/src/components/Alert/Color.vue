<template>
  <div class="space-y-6">
    <Alert>
      Default Alert
    </Alert>
    <Alert color="error">
      Error Alert
    </Alert>
    <Alert color="warn">
      Warning Alert
    </Alert>
    <Alert color="success">
      Success Alert
    </Alert>
    <Alert color="info">
      Info Alert
    </Alert>
    <Alert color="primary">
      Primary Alert
    </Alert>
  </div>
</template>
