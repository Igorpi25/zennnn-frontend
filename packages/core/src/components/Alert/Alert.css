.alert {
  @apply bg-white dark:bg-gray-700 rounded-md;
}

.alert__content {
  @apply flex-grow flex items-center text-sm leading-tight;
}

.alert__icon {
  @apply flex-shrink-0 pr-4;
}

.alert__close {
  @apply flex-shrink-0 ml-3 cursor-pointer select-none;
}

.alert__container {
  @apply flex rounded-md relative text-white py-3 px-4;
  @apply text-blue-500 bg-blue-500 bg-opacity-10;
}
.alert__container--closable {
  @apply pr-3;
}

.alert__container--primary {
  @apply text-blue-500 bg-blue-500 bg-opacity-10;
}
.alert__container--error {
  @apply text-white bg-red-900;
}
.alert__container--warn {
  @apply text-yellow-500 bg-yellow-500 bg-opacity-10;
}
.alert__container--success {
  @apply text-green-500 bg-green-500 bg-opacity-10;
}
.alert__container--info {
  @apply text-gray-100 bg-gray-500;
}

.alert__icon--error {
  @apply text-pink-500;
}
.alert__icon--info {
  @apply text-green-500;
}
