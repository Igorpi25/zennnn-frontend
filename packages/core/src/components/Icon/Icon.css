.icon {
  display: inline-flex;
  font-feature-settings: 'liga';
  justify-content: center;
  letter-spacing: normal;
  line-height: 1;
  text-indent: 0;
  vertical-align: middle;
  user-select: none;
}
.icon svg {
  text-indent: 0;
  letter-spacing: normal;
  line-height: 1;
  user-select: none;
  fill: currentColor;
  vertical-align: middle;
  width: 1em;
  height: 1em;
}
.icon--link {
  @apply cursor-pointer rounded;
  @apply focus:outline-none focus:ring focus:ring-blue-400;
}
.icon--link:focus:not(:focus-visible) {
  box-shadow: none;
  outline: 0;
}
