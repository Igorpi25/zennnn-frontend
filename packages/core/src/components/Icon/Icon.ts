import { h, computed, defineComponent } from 'vue'
import { convertToUnit } from 'vue-supp'

const SIZE_CLASS_MAP = {
  small: '18px',
  base: '24px',
  large: '32px',
}

export default defineComponent({
  name: 'Icon',

  props: {
    tag: {
      type: String,
      default: 'i',
    },
    size: {
      type: [Number, String],
      default: undefined,
    },
    small: Boolean,
    large: Boolean,
    base: {
      type: Boolean,
      default: true,
    },
    left: Boolean,
    right: Boolean,
  },

  setup(props, { attrs, slots }) {
    const fontSize = computed((): string | undefined => {
      return props.size
        ? convertToUnit(props.size)
        : props.small
        ? SIZE_CLASS_MAP.small
        : props.large
        ? SIZE_CLASS_MAP.large
        : props.base
        ? SIZE_CLASS_MAP.base
        : undefined
    })

    const genSvg = () => {
      const node = slots.default?.() ?? []
      const children = node[0] && node[0].children
      const icon = typeof children === 'string' ? children : undefined

      if (!icon) return undefined

      const svgData: Record<string, any> = {
        xmlns: 'http://www.w3.org/2000/svg',
        viewBox: '0 0 24 24',
        height: fontSize.value || '24',
        width: fontSize.value || '24',
        role: 'img',
        'aria-hidden': true,
      }
      if (fontSize.value) {
        svgData.style = {
          fontSize: fontSize.value,
          height: fontSize.value,
          width: fontSize.value,
        }
      }

      return h(
        'svg',
        svgData,
        h(
          'g',
          {
            fill: 'currentColor',
          },
          h('path', {
            d: icon,
          })
        )
      )
    }

    const clickable = attrs.onClick

    return () => {
      const tag = clickable ? 'button' : props.tag
      return h(
        tag,
        {
          class: {
            icon: true,
            'icon--left': props.left,
            'icon--right': props.right,
            'icon--link': clickable,
          },
          role: clickable ? 'button' : undefined,
          'aria-hidden': !clickable,
        },
        genSvg()
      )
    }
  },
})
