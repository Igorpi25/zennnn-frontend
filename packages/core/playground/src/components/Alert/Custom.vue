<template>
  <div class="space-y-6">
    <Alert :icon="false"> Primary Alert without icon </Alert>
    <Alert color="info">
      <template v-slot:icon>
        <Icon class="text-blue-500 mr-3">
          {{ ziCheckFill }}
        </Icon>
      </template>
      Info Alert custom icon
    </Alert>
    <div v-if="!model" class="text-center">
      <Btn @click="model = true"> Reset </Btn>
    </div>
    <Alert
      v-model="model"
      close
      transition="fade-transition"
      container-class="items-center text-white bg-green-500 bg-opacity-75 p-6"
      content-class="text-black text-2xl"
      icon-class="text-red-500"
    >
      Custom alert
      <template v-slot:close>
        <Btn
          x-small
          :primary="false"
          class="text-white bg-gray-600 hover:bg-gray-400 active:bg-gray-400"
        >
          Close
        </Btn>
      </template>
    </Alert>
  </div>
</template>

<script setup>
import { ref } from 'vue'
import { ziCheckFill } from '@zennnn/icons'

const model = ref(true)
</script>
