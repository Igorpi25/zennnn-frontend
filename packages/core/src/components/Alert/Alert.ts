import {
  h,
  computed,
  withDirectives,
  Transition,
  vShow,
  cloneVNode,
  defineComponent,
  PropType,
} from 'vue'

import { useModel, convertToUnit } from 'vue-supp'

import { ziCloseDelete, ziInfoBig } from '@zennnn/icons'

import Icon from '../Icon'

export default defineComponent({
  name: 'Alert',

  props: {
    modelValue: {
      type: [String, Boolean],
      default: true,
    },
    color: {
      type: String,
      validator: (value: string) => {
        return ['primary', 'error', 'warn', 'success', 'info'].includes(value)
      },
      default: 'primary',
    },
    icon: {
      type: Boolean,
      default: true,
    },
    close: Boolean,
    text: String,
    transition: String,
    mode: String as PropType<'in-out' | 'out-in' | 'default'>,
    minWidth: [String, Number],
    maxWidth: [String, Number],
    iconClass: {
      type: String,
      default: '',
    },
    closeClass: {
      type: String,
      default: '',
    },
    contentClass: {
      type: String,
      default: '',
    },
    containerClass: {
      type: String,
      default: '',
    },
  },

  emits: ['update:modelValue'],

  setup(props, { slots }) {
    const internal = useModel(props, 'modelValue')

    const containerClasses = computed(() => {
      return {
        alert__container: true,
        'alert__container--primary': props.color === 'primary',
        'alert__container--error': props.color === 'error',
        'alert__container--warn': props.color === 'warn',
        'alert__container--success': props.color === 'success',
        'alert__container--info': props.color === 'info',
        'alert__container--closable': props.close,
        [props.containerClass.trim()]: true,
      }
    })

    const styles = computed(() => {
      const styles: Record<string, unknown> = {}

      if (props.maxWidth) styles.maxWidth = convertToUnit(props.maxWidth)
      if (props.minWidth) styles.minWidth = convertToUnit(props.minWidth)

      return styles
    })

    const hasIcon = computed(() => {
      return slots.icon || props.icon
    })

    const hasClose = computed(() => {
      return slots.close || props.close
    })

    const close = () => {
      internal.value = false
    }

    const genIcon = () => {
      if (slots.icon) {
        return slots.icon()
      } else if (props.icon) {
        return h(
          Icon,
          {
            size: 24,
            class: {
              alert__icon: true,
              'alert__icon--error': props.color === 'error',
              'alert__icon--info': props.color === 'info',
              [props.iconClass.trim()]: true,
            },
          },
          {
            default: () => ziInfoBig,
          }
        )
      }
      return null
    }

    const genClose = () => {
      const closeChildren = slots.close?.() ?? []
      if (closeChildren.length === 1) {
        return cloneVNode(closeChildren[0], { onClick: close })
      } else if (props.close) {
        return h(
          Icon,
          {
            size: 24,
            class: {
              alert__close: true,
              [props.closeClass.trim()]: true,
            },
            onClick: close,
          },
          {
            default: () => ziCloseDelete,
          }
        )
      }
      return null
    }

    const genContent = () => {
      const children = []

      if (hasIcon.value) {
        children.push(genIcon())
      }

      children.push(
        h(
          'div',
          {
            class: {
              alert__content: true,
              [props.contentClass.trim()]: true,
            },
          },
          slots.default?.() ?? props.text
        )
      )

      if (hasClose.value) {
        children.push(genClose())
      }

      return withDirectives(
        h(
          'div',
          {
            class: 'alert',
            style: styles.value,
            role: 'alert',
          },
          h(
            'div',
            {
              class: containerClasses.value,
            },
            children
          )
        ),
        [[vShow, internal.value]]
      )
    }

    return {
      internal,
      genContent,
    }
  },

  render() {
    const content = this.genContent()

    if (this.transition) {
      return h(
        Transition,
        {
          name: this.transition,
          mode: this.mode,
        },
        { default: () => content }
      )
    } else {
      return this.internal ? content : undefined
    }
  },
})
