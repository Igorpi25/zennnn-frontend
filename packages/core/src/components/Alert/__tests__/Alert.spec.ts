import { h } from 'vue'
import { mount, VueWrapper } from '@vue/test-utils'
import { ziCheckFill } from '@zennnn/icons'

import Alert from '../Alert'
import Icon from '../../Icon'

describe('Alert.ts', () => {
  let mountFunction: (options?: Record<string, unknown>) => VueWrapper<any>

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(Alert, {
        ...options,
      })
    }
  })

  // Render

  it('should render component with default slot and match snapshot', () => {
    const wrapper = mountFunction({
      slots: {
        default: () => 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with text prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "error" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        color: 'error',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "warn" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        color: 'warn',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "success" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        color: 'success',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "info" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        color: 'info',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "primary" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        color: 'primary',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with close prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        close: true,
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom class props and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        close: true,
        closeClass: 'custom-close-class',
        iconClass: 'custom-info-icon-close',
        contentClass: 'content-custom-class',
        class: 'custom-class',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with icon slot and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        close: true,
        text: 'Alert',
      },
      slots: {
        icon: () => h(Icon, {
          class: 'text-blue-500 pr-4',
        }, { default: () => ziCheckFill }),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component without icon and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        icon: false,
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with close slot and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        text: 'Alert',
      },
      slots: {
        close: () => h('button', { class: 'custom-close' }, 'Close'),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit false modelValue on close icon click', async () => {
    const wrapper = mountFunction({
      props: {
        close: true,
        text: 'Alert',
      },
    })

    await wrapper.find('.alert__close').trigger('click')

    expect(wrapper.emitted()['update:modelValue'][0]).toEqual([false])
  })

  it('should toggle model value', async () => {
    const wrapper = mountFunction({
      props: {
        modelValue: false,
      },
      slots: {
        default: 'Alert',
      },
    })

    expect(wrapper.html()).toBe('<!---->')

    await wrapper.setProps({ modelValue: true })

    expect(wrapper.classes('alert')).toBe(true)
  })

  it('should set min-width with minWidth prop', () => {
    const wrapper = mountFunction({
      props: {
        minWidth: 100,
      },
    })

    expect((wrapper.element as HTMLElement).style.minWidth).toBe('100px')
  })

  it('should set max-width with maxWidth prop', () => {
    const wrapper = mountFunction({
      props: {
        maxWidth: 100,
      },
    })

    expect((wrapper.element as HTMLElement).style.maxWidth).toBe('100px')
  })
})
