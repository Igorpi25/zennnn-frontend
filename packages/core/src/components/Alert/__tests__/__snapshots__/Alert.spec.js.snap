// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert.js should render component with "error" color and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--error"><i class="icon alert__icon alert__icon--error" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with "info" color and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--info"><i class="icon alert__icon alert__icon--info" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with "primary" color and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--primary"><i class="icon alert__icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with "success" color and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--success"><i class="icon alert__icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with "warn" color and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--warn"><i class="icon alert__icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with close prop and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--primary alert__container--closable"><i class="icon alert__icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div><button class="icon icon--link alert__close" role="button" aria-hidden="false"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 10.586L8.466 7.05 7.051 8.465 10.586 12l-3.535 3.536 1.414 1.414 3.536-3.536 3.535 3.536 1.414-1.414L13.415 12l3.535-3.535-1.414-1.415-3.535 3.536z"></path>
        </g>
      </svg></button>
  </div>
</div>
`;

exports[`Alert.js should render component with close slot and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--primary"><i class="icon alert__icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div><button class="custom-close">Close</button>
  </div>
</div>
`;

exports[`Alert.js should render component with custom class props and match snapshot 1`] = `
<div class="alert custom-class" role="alert">
  <div class="alert__container alert__container--primary alert__container--closable"><i class="icon alert__icon custom-info-icon-close" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content content-custom-class">Alert</div><button class="icon icon--link alert__close custom-close-class" role="button" aria-hidden="false"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 10.586L8.466 7.05 7.051 8.465 10.586 12l-3.535 3.536 1.414 1.414 3.536-3.536 3.535 3.536 1.414-1.414L13.415 12l3.535-3.535-1.414-1.415-3.535 3.536z"></path>
        </g>
      </svg></button>
  </div>
</div>
`;

exports[`Alert.js should render component with default slot and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--primary"><i class="icon alert__icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with icon slot and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--primary alert__container--closable"><i class="icon text-blue-500 pr-4" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zm.646-6.535a1.25 1.25 0 01-1.805.01l-3.567-3.683a1.25 1.25 0 111.796-1.74l2.659 2.746 4.442-4.689a1.25 1.25 0 111.815 1.72z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div><button class="icon icon--link alert__close" role="button" aria-hidden="false"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 10.586L8.466 7.05 7.051 8.465 10.586 12l-3.535 3.536 1.414 1.414 3.536-3.536 3.535 3.536 1.414-1.414L13.415 12l3.535-3.535-1.414-1.415-3.535 3.536z"></path>
        </g>
      </svg></button>
  </div>
</div>
`;

exports[`Alert.js should render component with text prop and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--primary"><i class="icon alert__icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12 6.045a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.101c0-.658.56-1.192 1.25-1.192s1.25.534 1.25 1.192v5.87c0 .658-.56 1.192-1.25 1.192s-1.25-.534-1.25-1.192v-5.87zM1.2 12C1.2 6.035 6.036 1.2 12 1.2c5.965 0 10.8 4.835 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zM12 2.8a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component without icon and match snapshot 1`] = `
<div class="alert" role="alert">
  <div class="alert__container alert__container--primary">
    <div class="alert__content">Alert</div>
  </div>
</div>
`;
